aiohttp/web_urldispatcher.py

```diff
@@ -1028,44 +1099,65 @@ def register_resource(self, resource: AbstractResource) -> None:
         if self.frozen:
             raise RuntimeError("Cannot register a resource into frozen router.")
```

Comment:
These checks can be skipped if adding a resource, when there is already a group present. In that case, it only runs `_register_named_resource()`.

Maybe it would be better to put these checks in `_register_named_resource()` and call it unconditionally? Changing to `if not name: return` within that function.

Reply:
Maybe raise ValueError?